### PR TITLE
PHOENIX-3633 null pointer exception when subsquery for not exists ret…

### PR DESCRIPTION
**Pull request by stoty**
…urns empty result set

Co-authored-by: Jerry He <jerryjch@apache.org>

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/join/SubqueryIT.java [hunk 1]
```diff
@@ -528,10 +528,29 @@ public void testExistsSubquery() throws Exception {
...
-            
+
             rs = conn.createStatement().executeQuery("EXPLAIN " + query);
             String plan = QueryUtil.getExplainPlan(rs);
             assertPlansMatch(plans[2], plan);
+
+            //PHOENIX-3633
+            query = "SELECT * FROM " + tableName4 + " o WHERE NOT EXISTS (SELECT 1 FROM " + tableName1 + " i WHERE \"item_id\" = 'does not exist')";
+            statement = conn.prepareStatement(query);
+            statement = conn.prepareStatement(query);
```
Comment (virajjasani) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/join/SubqueryIT.java:539]: nit: repeated

File: phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java [hunk 2]
```diff
@@ -1217,8 +1217,10 @@ public boolean visitEnter(ExistsParseNode node) throws SQLException {
     @Override
     public Expression visitLeave(ExistsParseNode node, List<Expression> l) throws SQLException {
         LiteralExpression child = (LiteralExpression) l.get(0);
-        PhoenixArray array = (PhoenixArray) child.getValue();
-        return LiteralExpression.newConstant(array.getDimensions() > 0 ^ node.isNegate(), PBoolean.INSTANCE);
+        boolean exists = child != null
```
Comment (virajjasani) [inline phoenix-core/src/main/java/org/apache/phoenix/compile/ExpressionCompiler.java:1220]: nit: could be `elementExists`? It's upto you.

**Review by virajjasani — COMMENTED** [review]
Minor nits, looks good otherwise

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/join/SubqueryIT.java (hunk 1 above)
Comment (stoty) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/join/SubqueryIT.java:539]: oops

**Comment by stoty** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 28s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -1 :x: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ 4.x Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  13m 31s |  4.x passed  |
| +1 :green_heart: |  compile  |   1m  3s |  4.x passed  |
| +1 :green_heart: |  checkstyle  |   0m 39s |  4.x passed  |
| +1 :green_heart: |  javadoc  |   0m 44s |  4.x passed  |
| +0 :ok: |  spotbugs  |   2m 57s |  phoenix-core in 4.x has 962 extant spotbugs warnings.  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m  0s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 40s |  phoenix-core: The patch generated 15 new + 508 unchanged - 14 fixed = 523 total (was 522)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  javadoc  |   0m 44s |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   3m  9s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 131m 33s |  phoenix-core in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   | 163m 25s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1070/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/phoenix/pull/1070 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs hbaseanti checkstyle compile |
| uname | Linux 5413c7af3aea 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev/phoenix-personality.sh |
| git revision | 4.x / 3a3311c |
| Default Java | Private Build-1.8.0_242-8u242-b08-0ubuntu3~16.04-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1070/1/artifact/yetus-general-check/output/diff-checkstyle-phoenix-core.txt |
|  Test Results | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1070/1/testReport/ |
| Max. process+thread count | 6085 (vs. ulimit of 30000) |
| modules | C: phoenix-core U: phoenix-core |
| Console output | https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1070/1/console |
| versions | git=2.7.4 maven=3.3.9 spotbugs=4.1.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**